OpenRA.Mods.Common/Traits/AmmoPool.cs

```diff
@@ -0,0 +1,138 @@
+#region Copyright & License Information
+/*
+ * Copyright 2007-2015 The OpenRA Developers (see AUTHORS)
+ * This file is part of OpenRA, which is free software. It is made
+ * available to you under the terms of the GNU General Public License
+ * as published by the Free Software Foundation. For more information,
+ * see COPYING.
+ */
+#endregion
+
+using System.Collections.Generic;
+using System.Linq;
+using OpenRA.Traits;
+
+namespace OpenRA.Mods.Common.Traits
+{
+	[Desc("Actor has a limited amount of ammo, after using it all the actor must reload in some way.")]
+	public class AmmoPoolInfo : ITraitInfo
+	{
+		[Desc("Name of this ammo pool, used to link armaments to this pool.")]
+		public readonly string Name = "primary";
+
+		[Desc("How much ammo does this pool contain when fully loaded.")]
+		public readonly int Ammo = 1;
+
+		[Desc("Initial ammo the actor is created with. Defaults to Ammo.")]
+		public readonly int InitialAmmo = -1;
+
+		[Desc("Defaults to value in Ammo. 0 means no visible pips.")]
+		public readonly int PipCount = -1;
+
+		[Desc("PipType to use for loaded ammo.")]
+		public readonly PipType PipType = PipType.Green;
+
+		[Desc("PipType to use for empty ammo.")]
+		public readonly PipType PipTypeEmpty = PipType.Transparent;
+
+		[Desc("How much ammo is reloaded after a certain period.")]
+		public readonly int ReloadCount = 1;
+
+		[Desc("Sound to play for each reloaded ammo magazine.")]
+		public readonly string RearmSound = null;
+
+		[Desc("Time to reload per ReloadCount on airfield etc.")]
+		public readonly int ReloadTicks = 25 * 2;
+
+		[Desc("Whether or not ammo is replenished on its own.")]
+		public readonly bool SelfReloads = false;
+
+		[Desc("Time for each ReloadCount to reload automatically.")]
+		public readonly int SelfReloadTicks = 25 * 2;
```

Comment:
This description should be more consistent with the one of `ReloadTicks`. Actually, copy the fist part from there.